src/ds3.c

```diff
@@ -453,6 +453,20 @@ void ds3_request_set_max_keys(ds3_request* _request, uint32_t max_keys) {
     _set_query_param(_request, "max-keys", max_keys_s);
 }
 
+void ds3_request_set_preferred_number_of_chunks(ds3_request* _request, uint32_t num_chunks) {
+    char num_chunks_s[11];
```

Comment:
What does this magic number represent?

I'm assuming its the length of 2^32 (4294967296) + NULL, which you should go ahead and declare:
static char* UNSIGNED_LONG_BASE_10 = "4294967296"
static unsigned char UNSIGNED_LONG_BASE_10_STR_LEN = 11;